tests/spirv/address-space-specialize.slang

```diff
@@ -7,7 +7,10 @@
 static int gArray0[2];
 groupshared int gArray1[2];
 
-// CHECK: %array = OpFunctionParameter %_ptr_Private__arr_int_int_2
+// Note: static globals are inside a context variable
+// CHECK: OpTypeArray %int %int_2
+// CHECK: OpVariable %_ptr_Function__arr_int_int_2 Function
```

Comment:
What is "a context variable"?
Why does it change the storage type from private to function here?

Reply:
After reading the all implementation, I learned that your change is expected to change from private to function.
But I am still not sure what "context variable" is.

Reply:
>What is "a context variable"?

Sometimes Slang is provided global variables or needs to track some state of a variable "globally". The problem is that some targets cannot use global variables for various reasons.

To solve this problem:
1. Slang allocates all "global" variables in the entry-point of a program. "global" variables are now local variables which are only visible to the current invocation.
2. Slang has a **"context variable"** which is passed to (most) functions in Slang. This Composite variable stores a pointer to all "global" variables, this allows all function-calls to access "global" variables. This context variable is only visible to the current invocation


> Why does it change the storage type from private to function here?

Since we use a "context variable" in this scenario we store "global" variables as a local variable only visible to the current invocation. This means we require the storage class of [`Function`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Storage_Class) for the `OpVariable` definition here.

Reply:
I guess this is same as the context I saw in Metal legalization.
I am wondering if it will cause any problems regarding the memory pointer.
Our user guide says the following,
```
Slang supports pointers to global memory, but not shared or local memory. For example, it is invalid to define a pointer to a local variable.
```
https://shader-slang.com/slang/user-guide/convenience-features.html#pointers-limited

I am not clear on why we don't support a memory pointer to a local variable. I assume that SPIR-V doesn't support it.
Your change will move the pointers from global memory to local/function memory.
And I am not sure if it will cause a problem.

Reply:
SPIR-V has local pointers.
Looking at SPIR-V spec I don't see anything that could break our use of global's.

@csyonghe, may I ask for clarification on this topic to make-sure we are not missing anything?